### PR TITLE
Holds on in progress

### LINKED ISSUE #268: holds on items that are in progress do not work
should notify user when creating a hold on an item that is already picked up

### PR DESCRIPTION
**Pull request by tomecho**
fixes #268 
and an error we never thought of where holds consider any rental between the date ranges to be a conflict

### CODE REVIEW

**Comment by tomecho** [comment]
image
Example of what happens when an in progress item is put on hold

File: app/controllers/holds_controller.rb [hunk 1]
```diff
@@ -18,6 +19,12 @@ def create
     if @hold.save
       flash[:success] = 'Hold Successfully Created'
       @hold.handle_conflicting_rentals
+
+      # rental that has already started and conflicts with hold, needs to be handled
+      # by a csr
+      if rental = @hold.conflicting_ongoing_rental
+        flash_message :warning, "Ongoing rental with item #{url_for rental}".html_safe
```
Comment (josephkelley) [inline app/controllers/holds_controller.rb]: Let's make the URL here more user friendly. Maybe make "item" clickable with a link to the `url_for`?

File: app/controllers/holds_controller.rb [hunk 2]
```diff
@@ -31,6 +38,12 @@ def update
     if @hold.update(hold_params)
       flash[:success] = 'Hold Successfully Updated'
       @hold.handle_conflicting_rentals
+
+      # rental that has already started and conflicts with hold, needs to be handled
+      # by a csr
+      if rental = @hold.conflicting_ongoing_rental
+        flash_message :warning, "Ongoing rental with item #{url_for rental}".html_safe
```
Comment (josephkelley) [inline app/controllers/holds_controller.rb]: See above.

**Comment by tomecho** [comment]
@josephkelley plz rereview

**Pull request merged**